yarn-project/p2p/src/attestation_pool/memory_attestation_pool.ts

```diff
@@ -41,14 +45,27 @@ export class InMemoryAttestationPool implements AttestationPool {
     return Promise.resolve();
   }
 
+  public deleteAttestationsForSlotAndProposal(slot: bigint, proposalId: string): Promise<void> {
```

Comment:
When would we deleting for one `slot|proposalId` combination and not just `slot`? I guess for a reorg, but it seems like when we would keep track of things until it is in proven?

Reply:
The granularity is there for granularities sake